src/go/k8s/apis/redpanda/v1alpha1/console_types.go

```diff
@@ -0,0 +1,189 @@
+// Copyright 2022 Redpanda Data, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
+
+package v1alpha1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+// ConsoleSpec defines the desired state of Console
+type ConsoleSpec struct {
+	// +optional
+	// +kubebuilder:default=console
+	MetricsNamespace string `json:"metricsNamespace"`
+
+	// +optional
+	// +kubebuilder:default=true
+	ServeFrontend bool `json:"serveFrontend"`
+
+	// +optional
+	Server Server `json:"server"`
+
+	Schema        Schema                 `json:"schema"`
+	ClusterKeyRef corev1.ObjectReference `json:"clusterKeyRef"`
+	Deployment    Deployment             `json:"deployment"`
+	Connect       Connect                `json:"connect"`
+}
+
+// Server is the Console app HTTP server config
+type Server struct {
+	// +kubebuilder:default="30s"
+	ServerGracefulShutdownTimeout string `json:"gracefulShutdownTimeout,omitempty" yaml:"gracefulShutdownTimeout,omitempty"`
+
+	HTTPListenAddress string `json:"listenAddress,omitempty" yaml:"listenAddress,omitempty"`
```

Comment:
is this just port or... ? I assume you don't know the IP?

Reply:
I think this is the host of the HTTP server [REF](https://github.com/cloudhut/common/blob/775e4e1c9e0304afde6397dd61840d5443541bdd/rest/server.go#L68)